lib/functions/froxlor/function.generateDKIM.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+	function generateDkim( $vars ) {
+
+		if ( Settings::Get( 'dkim.use_dkim' ) !== '1'
+			|| !isset( $vars['dkim_pubkey'] )
+			|| $vars['dkim_pubkey'] == '' ) {
+				return false;
+		}
+		
+		$selector = "custom";
+		
+		// start
+		$dkim_txt = 'v=DKIM1;';
+
+		// algorithm
+		$algorithm = explode( ',', Settings::Get( 'dkim.dkim_algorithm' ) );
+		$alg = '';
+
+		foreach ( $algorithm as $a ) {
+			if ( $a == 'all' ) {
+				break;
+			} else {
+				$alg .= $a . ':';
+			}
+		}
+
+		if ( $alg != '' ) {
+			$alg = substr( $alg, 0, -1 );
+			$dkim_txt .= 'h=' . $alg . ';';
+		}
+
+		// notes
+		if ( trim( Settings::Get( 'dkim.dkim_notes' ) != '' ) ) {
+			$dkim_txt .= 'n=' . trim( Settings::Get( 'dkim.dkim_notes' ) ) . ';';
+		}
+
+		// key
+		$dkim_txt .= 'k=rsa;p=' . trim( preg_replace( '/-----BEGIN PUBLIC KEY-----(.+)-----END PUBLIC KEY-----/s', '$1', str_replace( "\n", '', $vars['dkim_pubkey'] ) ) ) . ';';
+
+		// service-type
+		if ( Settings::Get( 'dkim.dkim_servicetype' ) == '1' ) {
+			$dkim_txt .= 's=email;';
+		}
+
+		// end-part
+		$dkim_txt .= 't=s';
+
+		// split if necessary
+		$dkim_record = '';
+		$lbr = 50;
+		for ( $pos = 0; $pos <= strlen( $dkim_txt ) -1; $pos += $lbr ) {
+			$dkim_record .= ( ( $pos == 0 ) ? '( "' : "\t\t\t\t\t \"" ) . substr( $dkim_txt, $pos, $lbr ) . ( ( $pos >= strlen( $dkim_txt ) - $lbr ) ? '" )' : '"' );
+		}
+
+		// Host entry
+		if ( !empty( $selector ) ) {

```

Comment:
Thanks! I'll check on that. The "custom" selector doesn't have to be there. It just made sense to me because that's how I had it set up on my server - [servername].domainkey.[domain]. Maybe it could be a dynamic option under Settings -> DomainKeys?